src/generator.cs

```diff
@@ -4096,6 +4096,9 @@ void GenerateArgumentChecks (MethodInfo mi, bool null_allowed_override, Property
 		if (null_allowed_override)
 			return;
 
+		if (AttributeManager.HasAttribute<NullAllowedAttribute> (mi))
+			ErrorHelper.Show (new BindingException (1118, false, $"[NullAllowed] should not be used on methods, like '{mi}', but only on properties, parameters and return values."));
```

Comment:
Using `ErrorHelper.Warning` makes it clearer that this is a warning.

Reply:
@rolfbjarne that API does not exist (at least in the generator code base)

Reply:
It should :/ oh well, maybe some other day.